### DISCUSSION TITLE
Down velocity in INS/GNSS integration example using synthetic data

### DISCUSSION THREAD

**Discussion by yuzhangjun-gps** [rodralez/NaveGo#61] [Q&A] [answered]
Dear NaveGo-Team, thank you for your work, I do appreciate it.

Performances of the two simulated INS/GNSS systems are compared in the file navego_example_synth.m, but the down velocity seems to be problematic (Down velocity errors of the two simulated INS/GNSS systems are always outside the margin of error).

Down velocity error of IMU2:
![image](https://user-images.githubusercontent.com/83227706/116100362-6b1e5580-a6df-11eb-89eb-40e8824286e5.png)

**Comment by rodralez** [comment]
Hi,

First of all, sorry for the super delayed reply.

I cannot see this behavior in the current NaveGo version.

I get this figure for the velocity errors:

![image](https://user-images.githubusercontent.com/5432060/120046705-212fd480-bfe9-11eb-822f-cdbd8b8ea96b.png)

Nevertheless, I changed the way of simulating the inertial sensors some months ago and maybe this could cause a problem with the biases estimation.

I will check.

Best wishes.

**Reply by yuzhangjun-gps** [reply]
Hi,
This is the result of a recent run where the previously mentioned problem seems to have been unresolved. I am looking forward to your early reply.
![untitled](https://user-images.githubusercontent.com/83227706/129724797-d99e3f53-ec6a-4261-a1a5-1a457b43a6ea.jpg)
![untitled1](https://user-images.githubusercontent.com/83227706/129724805-69a9c0bc-5ef8-49e9-803b-40a7a878a707.jpg)

**Comment by rodralez** [comment, marked as answer]
Hi,

Thank you for pointing out this problem again.

I re checked my code and found an error in function F_update.m. There was a problem with the sign of matrices DCMbn corresponding to gyros dynamic biases in matrices F and G. 

Old matrices were:

    F = [F11  F12  F13  DCMbn   Z  ;
        F21  F22  F23   Z       DCMbn  ;
        F31  F32  F33   Z       Z      ;
        Z    Z    Z     Fgg     Z      ;
        Z    Z    Z     Z       Faa    ;
        ];
    
    G = [DCMbn  Z     Z    Z ;
        Z      DCMbn  Z    Z ;
        Z      Z      Z    Z ;
        Z      Z      Fbg    Z ;
        Z      Z      Z    Fba ;
        ];
    
And new matrices are:

    F = [F11  F12  F13  DCMbn   Z  ;
        F21  F22  F23   Z       -DCMbn  ;
        F31  F32  F33   Z       Z      ;
        Z    Z    Z     Fgg     Z      ;
        Z    Z    Z     Z       Faa    ;
        ];
    
    G = [DCMbn  Z     Z    Z ;
        Z      -DCMbn  Z    Z ;
        Z      Z      Z    Z ;
        Z      Z      Fbg    Z ;
        Z      Z      Z    Fba ;
        ];
     
Moreover, now the default method for simulating the dynamic bias noise in function noise_b_dyn.m is Aggarwal's formulation.

Please, download the last commit and let me know if everything is OK:

Best regards.

**Reply by yuzhangjun-gps** [reply]
Thanks for your answer @rodralez, the problem mentioned has been solved.
